http/transport.go

```diff
@@ -55,8 +58,25 @@ func splitAddr(host string) (string, string, int, error) {
 	return hostParts[0], hostParts[1], port, nil
 }
 
+func SetExternalHTTPProxy(proxy string) {
```

Comment:
Let's not modify http package since it is being used by server package and keep the configuration of proxy server simplier. We could pass a callback function through `server.RunEdgeHTTPProxyServer(...)` if you needs to.

Reply:
This is my first foray into go, so sorry if my questions don't make sense:
It seem we have to modify the http package at least slightly in order to add Proxy: argument, to the transport (current line 93).  We could define proxyForEdge in another package.  Is this what you are suggesting?  If so, is there a package you suggest?

Reply:
@zhucebuliaolongchuan I think it's fine to modify the http package here, since this setting applies to the all requests going to the cloud it would seem. As I suggested below though, it is better to just parse the URL in main once and change the signature of `EdgeTransport` to accept an optional external HTTP proxy URL argument.

Reply:
It seems like we are talking about the same thing but my sentence is kind of misleading here. We have to modify http package here for sure, but I am suggesting not to pass down the configuration directly like this.

Reply:
okay.  I understand now.  I should have an alternate solution later today.  Thanks for the feedback.

Reply:
@kylestein-arm after your latest update, this won't be needed I guess?

Reply:
Yes.  Forgot to remove it.  Thanks.

Reply:
fixed